lib/api/objectPut.js

```diff
@@ -99,19 +99,24 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
             // have overwritten if there was an existing version or object
             const oldByteLength = objMD && objMD['content-length']
```

Comment:
There are some edge-ish cases where the old byte length which was overwritten by the new object/version is not the content length of objMD (the latest version), i.e. when a null version put when versioning suspended is overwriting a previous null version that is not the latest version.

The logic used for this situation is located in versioningPreprocessing in certain conditionals. That function probably should be refactored at some point because it's gotten a bit long, but we might be able to return old byte length from there before we delete the old null version. Maybe we can add this as a todo item?